frontend/src/component/feature/StrategyTypes/FlexibleStrategy/StickinessSelect/StickinessSelect.tsx

```diff
@@ -0,0 +1,52 @@
+import Select from 'component/common/select';
+import { SelectChangeEvent } from '@mui/material';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+const builtInStickinessOptions = [
+    { key: 'default', label: 'default' },
+    { key: 'userId', label: 'userId' },
+    { key: 'sessionId', label: 'sessionId' },
+    { key: 'random', label: 'random' },
+];
+
+interface IStickinessSelectProps {
+    label: string;
+    value: string | undefined;
+    editable: boolean;
+    onChange: (event: SelectChangeEvent) => void;
+    dataTestId?: string;
+}
+export const StickinessSelect = ({
+    label,
+    editable,
+    value,
+    onChange,
+    dataTestId,
+}: IStickinessSelectProps) => {
+    const { context } = useUnleashContext();
+
+    const resolveStickiness = () =>
```

Comment:
comment or more intention revealing name would help as the first time code reader I don't know what those filters from context are doing

Reply:
If the code was here before I'd still improve it after touching it to leave the camp cleaner after ourselves